docs/src/main/asciidoc/security-customization.adoc

```diff
@@ -95,6 +92,65 @@ public class RolesAugmentor implements SecurityIdentityAugmentor {
 }
 ----
 
+Here is another example showing how to use the client certificate available in the current link:security-built-in-authentication#mutual-tls[Mutual TLS] request to add more roles:
+
+[source,java]
+----
+import java.security.cert.X509Certificate;
+import io.quarkus.security.credential.CertificateCredential;
+import io.quarkus.security.identity.AuthenticationRequestContext;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.identity.SecurityIdentityAugmentor;
+import io.quarkus.security.runtime.QuarkusSecurityIdentity;
+import io.smallrye.mutiny.Uni;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.function.Supplier;
+import java.util.Set;
+
+@ApplicationScoped
+public class RolesAugmentor implements SecurityIdentityAugmentor {
+
+    @Override
+    public int priority() {
+        return 0;
+    }
+
+    @Override
+    public Uni<SecurityIdentity> augment(SecurityIdentity identity, AuthenticationRequestContext context) {
+        return context.runBlocking(build(identity));
```

Comment:
I'd like to understand what's the point of moving this work to the worker pool, because everything we do here is in-memory and probably much faster than a thread context switch.

Reply:
@Ladicek Was a copy and paste, good point let me update

Reply:
@Ladicek Fixed it, also added a comment to the example I copied/pasted from

Reply:
Looks good, thanks!